tracing/src/main/java/com/palantir/tracing/Tracer.java

```diff
@@ -143,10 +141,15 @@ public static void fastCompleteSpan() {
      * Like {@link #fastCompleteSpan()}, but adds {@code metadata} to the current span being completed.
      */
     public static void fastCompleteSpan(Map<String, String> metadata) {
-        popCurrentSpan()
-                .filter(openSpan -> currentTrace.get().isObservable())
-                .map(openSpan -> toSpan(openSpan, metadata))
-                .ifPresent(Tracer::notifyObservers);
+        Trace trace = currentTrace.get();
+        if (trace != null) {
+            boolean observable = isTraceObservable();
+            Optional<OpenSpan> span = popCurrentSpan();
+            if (observable) {
```

Comment:
should we reuse the trace to avoid an additional thread local get?

```suggestion
            if (isObservable(trace)) {
```

Reply:
good call, I think we want `trace.isObservable()` rather than a new method unless I'm misunderstanding.

Reply:
All the call sites should have a separate null check on the trace, so I think this will be okay.